src/main/java/pl/put/poznan/transformer/logic/AbbreviationDecorator.java

```diff
@@ -0,0 +1,121 @@
+package pl.put.poznan.transformer.logic;
+
+import java.io.FileNotFoundException;
+import java.util.ArrayList;
+import java.io.File;
+import java.util.Scanner;
+
+public class AbbreviationDecorator extends BasicTextTransformer {
+    private final TextTransformer transformer;
+
+    public AbbreviationDecorator(TextTransformer wrappedTransformer) {
+        this.transformer = wrappedTransformer;
+    }
+
+    @Override
+    public String transform(String text) {
+        return abbreviation(transformer.transform(text));
+    }
+
+    public void readFile(ArrayList<String> abbreviations, ArrayList<String> fullWords) {
+        String[] wordsArray;
+        try {
+            File myObj = new File("src/main/resources/abbreviation.txt");
+            Scanner myReader = new Scanner(myObj);
+            while (myReader.hasNextLine()) {
+                String data = myReader.nextLine();
+                wordsArray = data.split("\t");
+                abbreviations.add(wordsArray[0]);
+                fullWords.add(wordsArray[1]);
+            }
+            myReader.close();
+        } catch (FileNotFoundException e) {
+            System.out.println("An error occurred.");
+        }
+    }
+
+    public String abbreviation(String text) {
+        String[] split = text.split(" ");
+        int splitLength = split.length;
+        var abbreviations = new ArrayList<String>();
+        var fullWords = new ArrayList<String>();
+        var toCapitalize = new ArrayList<Integer>();
+
+        readFile(abbreviations, fullWords);
+
+        var output = new ArrayList<String>();
+        int isAbbreviated = 0;
+        int i = 0;
+        int size = 0;
+        int iteration;
+        int capitalize = 0;
+        while (i < splitLength) {
+            iteration = 0;
+            for (String word : fullWords) {
+                if (word.length() == 0) {
+                    continue;
+                }
+                toCapitalize.clear();
+                String[] splitWords = word.split(" ");
```

Comment:
is this abbreviationWords?

Reply:
no nie do końca bo jest to rozwinięcie skrótów